### PR TITLE
AvailabilitySet refinements, javadocs, bug fixes

### PR DESCRIPTION
**Pull request by martinsawicki**

### CODE REVIEW

File: azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.microsoft.azure.management.resources;
 
 import com.microsoft.azure.management.resources.fluentcore.arm.Region;
-import com.microsoft.azure.management.resources.fluentcore.arm.models.Taggable;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
 import com.microsoft.azure.management.resources.fluentcore.model.*;
```
Comment (anuchandy) [inline azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java]: Check-style requires all models to be imported explicitly.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetsImpl.java [hunk 2]
```diff
@@ -71,13 +71,13 @@ public AvailabilitySet typeConvert(AvailabilitySetInner availabilitySetInner) {
...
     @Override
-    public AvailabilitySet get(String groupName, String name) throws CloudException, IOException {
+    public AvailabilitySetImpl get(String groupName, String name) throws CloudException, IOException {
         ServiceResponse<AvailabilitySetInner> response = this.client.get(groupName, name);
         return createFluentModel(response.getBody());
     }
 
     @Override
-    public AvailabilitySet.DefinitionBlank define(String name) {
+    public AvailabilitySetImpl define(String name) {
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetsImpl.java]: Why this change? Returning an `AvailabilitySetImpl` exposes all methods to the user at once.

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java [hunk 3]
```diff
@@ -162,15 +167,15 @@
          * @param dnsName the leaf domain label to use. This must follow the required naming convention for leaf domain names.
          * @return the next stage of the public IP address definition
```
Comment (jianghaolu) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java]:
```java
          * @return the next stage of the resource update
 ```

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java [hunk 4]
```diff
@@ -162,15 +167,15 @@
...
          */
-        T withLeafDomainLabel(String dnsName);
+        Update withLeafDomainLabel(String dnsName);
 
         /**
          * Ensures that no leaf domain label will be used. 
          * <p>
          * This means that this public IP address will not be associated with a domain name.
          * @return the next stage of the resource definition
```
Comment (jianghaolu) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java]: same

**Comment by jianghaolu** [comment]
Considering that you have work on top of this we should be able to merge this one and you can address the feedbacks later.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetsImpl.java (hunk 2 above)
Comment (martinsawicki) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetsImpl.java]: nope. remember, the user can only view this object through interfaces. The impl is package-internal. The impls should always return impls, not interfaces. Otherwise, we wouldn't be able to implement Update and Definition using the same impl.

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java (hunk 3 above)
Comment (martinsawicki) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java]: oops. fixed in a future checkin.

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java (hunk 4 above)
Comment (martinsawicki) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/PublicIpAddress.java]: good eye. fixed in a future checkin.

**Comment by martinsawicki** [comment]
sounds good.

File: azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java (hunk 1 above)
Comment (jianghaolu) [inline azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java]: Yes. This can probably be found in some sort of settings. In IntelliJ it's a threshold setting of how many classes in one package to import. Setting it to 99 instead of 6 gets the problem solved.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetsImpl.java (hunk 2 above)
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetsImpl.java]: Make sense.

**Comment by jianghaolu** [comment]
Merging.

File: azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java (hunk 1 above)
Comment (martinsawicki) [inline azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java]: note the import ".*" stuff is not part of my checkin. it was already there. I agree it should be explicit.

Comment (jianghaolu) [inline azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/ResourceGroup.java]: You're right. This is fixed on my local machine. Pending javadocs review.
